fetch/api/credentials/authentication-redirection.any.js

```diff
@@ -0,0 +1,26 @@
+// META: global=window,worker
+// META: script=/common/get-host-info.sub.js
+
+const authorizationValue = "Basic " + btoa("user:pass");
+async function getAuthorizationHeaderValue(url)
+{
+  const headers = { "Authorization": authorizationValue};
+  const requestInit = {"headers": headers};
+  const response = await fetch(url, requestInit);
+  return response.text();
+}
+
+promise_test(async test => {
+  const result = await getAuthorizationHeaderValue("/fetch/api/resources/dump-authorization-header.py");
+  assert_equals(result, authorizationValue);
+}, "getAuthorizationHeaderValue - no redirection");
+
+promise_test(async test => {
+  const result = await getAuthorizationHeaderValue("/fetch/api/resources/redirect.py?location=" + encodeURIComponent("/fetch/api/resources/dump-authorization-header.py"));
+  assert_equals(result, authorizationValue);
+}, "getAuthorizationHeaderValue - same origin redirection");
+
+promise_test(async (test) => {
+  const result = await getAuthorizationHeaderValue(get_host_info().HTTP_REMOTE_ORIGIN + "/fetch/api/resources/redirect.py?allow_headers=Authorization&location=" + encodeURIComponent(get_host_info().HTTP_ORIGIN + "/fetch/api/resources/dump-authorization-header.py"));
```

Comment:
Thank you for writing this test. 
I am implementing fix for Gecko in .
I tried these tests to verify my fix. Interestingly, we are not able to validate the test for service workers as we encounter an unhandled reject.
```
TypeError: NetworkError when attempting to fetch resource
```
Interestingly, the problem is not only limited to Gecko but also for [other browsers ](https://wpt.fyi/results/fetch/api/credentials/authentication-redirection.any.serviceworker.html?label=master&label=experimental&aligned&view=subtest&q=authentication-redirection.any). Interestingly, it fails even for Safari which has the fix for this issue.  For Safari. the [tests are passing for other sub-categories](https://wpt.fyi/results/fetch/api/credentials?label=master&label=experimental&aligned&view=subtest&q=authentication-redirection.any). 
Hence, I suspect the issue might be related to the service worker related tests. 
I would like to check with others who might be trying these tests.

Reply:
@MayyaSunil if you rename this file to `authentication-redirection.https.any.js` I suspect it'll start passing. Would you be willing to make that change as part of the Gecko fix? That is, I'm assuming the problem is that service workers require a secure context and we're not providing that.

Reply:
Hmm, although some subtests are passing. Not entirely sure what's up then. @youennf?

Reply:
> @MayyaSunil if you rename this file to `authentication-redirection.https.any.js` I suspect it'll start passing. Would you be willing to make that change as part of the Gecko fix? That is, I'm assuming the problem is that service workers require a secure context and we're not providing that.

@annevk you might be right as I say security warnings during the service workers test only. I will try your fix and let you know how it goes. 

btw, I dont see these tests in our official rep yet. I am not sure why the [moz-wptsync-bot](https://github.com/moz-wptsync-bot) has added the blocker label.

Reply:
Renaming the file to authentication-redirection.https.any.js did not help. However, I do not see the security warnings.

Reply:
Okay, so the remaining problem is likely that it's using an HTTP_ORIGIN and not an HTTPS_ORIGIN. I suspect there is an HTTPS equivalent but that might be worth double checking in the get-host-info file. Then it won't get blocked due to Mixed Content.

That still doesn't explain it entirely to me, but maybe the service worker test is always running over HTTPS and the rename isn't needed and so we were just running into Mixed Content? (But that wouldn't explain the security warning you saw, but maybe that was something else.)

Reply:
Thanks @annevk, after changing HTTP_ORIGIN to HTTPS_ORIGIN and HTTP_REMOTE_ORIGIN to HTTPS_REMOTE_ORIGIN the issue got resolved. There is no need to rename the file. 
I can submit the patch if these tests are merged into our repo. I need check with someone in our team about why these tests are not merged yet.

Reply:
Thanks @annevk, after changing HTTP_REMOTE_ORIGIN to HTTPS_REMOTE_ORIGIN and HTTP_ORIGIN to HTTPS_ORIGIN, the issue seems to have fixed. I will submit the fix once these tests are merged into our repo.